src/components/notes/editor.jsx

```diff
@@ -13,196 +7,58 @@ import EditorLoading from 'components/notes/editor-loading'
 import EditorLoadingError from 'components/notes/editor-loading-error'
 import SharingWidget from 'components/notes/sharing'
 import BackFromEditing from 'components/notes/back_from_editing'
-
 import IsPublicContext from 'components/IsPublicContext'
 
-import CollabProvider from 'lib/collab/provider'
-import ServiceClient from 'lib/collab/stack-client'
-
-import {
-  getShortNameFromClient,
-  getParentFolderLink,
-  getAppFullName
-} from 'lib/utils.js'
+import useNote from 'hooks/useNote'
+import useServiceClient from 'hooks/useServiceClient'
+import useCollabProvider from 'hooks/useCollabProvider'
+import useTitleChanges from 'hooks/useTitleChanges'
+import useForceSync from 'hooks/useForceSync'
+import useReturnUrl from 'hooks/useReturnUrl'
+import useUser from 'hooks/useUser'
```

Comment:
Lot of hooks, I would be interested to know if they are really decoupled and usable in a standalone manner, each one from each one. At this point I am wondering if a class component would not be better than many decoupled hooks.

Reply:
Most of them are pretty specific to this component will probably never be used in another context.

Reply:
What I mean was : all this hooks have to be used in combination and it seems that we gain nothing with having them in different files since they are very coupled in functionality. Could there be only one hook useEditor or useNote doing all these things ?

Reply:
I have mixed feelings about that. My initial purpose was to explode all logic in different files in order to avoid a unique very long file.

The Editor does not have really more than that. If we put everything in one unique hook, I'm afraid that we'll end un with exactly what we had before this PR.

Reply:
OK.

> My initial purpose was to explode all logic in different files in order to avoid a unique very long file.

I think the long file was not the problematic thing before, it was the super long render method. 

> If we put everything in one unique hook, I'm afraid that we'll end un with exactly what we had before this PR.

Extracting each hook and keeping them in the same file would have been OK for me. I personnally prefer when highly cohesive methods are in the same file, I find it easier to refactor and see the overall picture. However when things are not cohesive, sure, separate files are neded.

Here, Editor has to have knowledge of how to assemble all the little hooks, I have trouble saying why, but it troubles me that the orchestration of all those hooks is done in the render.